modules/Auth/include/AuthHandler.hpp

```diff
@@ -227,9 +263,12 @@ class AuthHandler {
     std::map<int, std::unique_ptr<EVSEContext>> evses;
 
     std::list<int> plug_in_queue;
-    std::set<std::string> tokens_in_process;
+    std::set<ProvidedIdToken> tokens_in_process;
     std::condition_variable cv;
+    std::condition_variable processing_finished_cv;
     std::mutex event_mutex;
+    std::mutex withdraw_mutex;
```

Comment:
this could probably also use the event_mutex now, or am I mistaken?

Reply:
Initially I thought so too. If we only use the `event_mutex` multiple withdraw requests could run in parallel and they could override `this->withdraw_request` in that case, because the `event_mutex` is released [here](https://github.com/EVerest/everest-core/pull/992/files#diff-8db9b5c55c5494b78d2e0be7e5aa44d64c9361a2862b47ae6f13a4cd86a63baaR894) while waiting for the targeted processing threads to finish. It ensures that only one withdraw_request can run at a time. Since a withdraw request should not block for a significant amount of time this seems to be the best solution to me